doc/adr/0013-flame-graph-support.md

```diff
@@ -0,0 +1,132 @@
+
+![current](0013/flame-graph-kernel-0.0.1.png)
+
+There are 2 possibilites to support the flame graph in the TSP.
+
+### Solution 1: New flame graph data provider type
+
+The view it's self is very similar to data providers of type `TIME_GRAPH`. The main difference is that the x-axis is the duration and not time. The Trace Compass back-end currently has such an data provider available with the same API than `TIME_GRAPH` data providers. However, this data provider is not exposed to the server through the `getDescriptor()` method of the corresponing data provider factory. It will need a new type to be added to the `ProviderType` enum.
+
+With this solution similar endpoints can be defined which have the advantage that the endpoints and data structures are known. Back-end filtering and highlighting would be supported out of the box. Assuming the new `Provider Type` is called `gantt` the following endpoints will be defined:
+
+```
+    POST /experiments/{expUUID}/outputs/{outputId}/styles
+    Get the styles defined for this data provider
+
+    GET /experiments/{expUUID}/outputs/{outputId}/annotations
+    Get list of annotation categories for this view
+
+    POST /experiments/{expUUID}/outputs/{outputId}/annotations
+    Get list of annotation for a given annotation categories for this view
+
+    POST /experiments/{expUUID}/outputs/gantt/{outputId}/tree
+    Returns the gantt chart entry model
+
+    POST /experiments/{expUUID}/outputs/gantt/{outputId}/states
+    Returns the states from the for given entries
+
+    POST /experiments/{expUUID}/outputs/gantt/{outputId}/tooltip
+    Get detailed information of a state, annotation or arrow
+```
+
+**Note**:
+
+- This solution will allow us to fetch data depending on the zoom level. Since the flame graph data is limited in size (in comparison to regular time graphs), all the data can fetched once and all the zooming is done in FE only.
+- The annotation endpoints and styles endpoint are independent of the chart type (`TIME_GRAPH`) and can be provided for any chart type. 
+- The endpoints and data structures for the time based gantt chart (providerType `TIME_GRAPH`) are designed around timestamps. For example, states have a start and end, or annotations have a `start` and `duration`, or query parameter `requested_time_range` is meant for time stamps. Those names should be more generic, i.e. instead of `duration` it should be `delta`, instead of `requested_time_range` it should be `requested_range` and so on. Having generic names it won't be necessary to duplicate endpoints, data structes etc. for different x-axis.  
+- To be able to get a flame graph for a given time range of the trace the `requested_time_range` should be used for that and a new query parameter `requested_range` for the view range (see bullet above).
+- Arrows for flame chart won't make make sense and don't need to be supported.
```

Comment:
Maybe not for flame chart but perhaps for other uses of this new provider type?

Reply:
Yes, flame graph. I'll update it.